tools/codegen/api/cpp.py

```diff
@@ -252,16 +252,20 @@ def argument_not_this(
 
 def argument(
     a: Union[Argument, TensorOptionsArguments, ThisArgument],
+    is_out_argument: bool,
```

Comment:
Should this not be a property (maybe synthetic) of `Argument`? It seems odd to have it in a sidecar like this - e.g. it's nonsense for it to be `True` if `a` is a `TensorOptionsArguments` or `ThisArgument`.